.github/workflows/validate.yml

```diff
@@ -0,0 +1,84 @@
+# Github Actions workflow that uses SFDX to authenticate and run validations and deployments
+name: validate
+on:
+  # https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - dev
+      - master
+
+# NOTE: ENV vars can be declated globally here and shared among jobs/steps below
+# See: https://github.com/actions/starter-workflows/blob/master/ci/google.yml
+env:
+  AUTH_FILE_KEY: ${{ secrets.AUTH_FILE_KEY }}
+
+jobs:
+  # TODO: Add workflow that creates release tag on successful merge into Master
+  authenticate_to_prod:
+    runs-on: ubuntu-latest
+    name: Run unit tests
+    steps:
+      # Using v2 in this example ensures you're using a stable version of the checkout action.
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Check for existence of files or env vars
+      - name: Check required items
+        run: |
+          if [ -z ${{ env.AUTH_FILE_KEY }} ]; then
+            echo "Missing Auth File Key in Github secrets! Exiting!"
+            exit 1
+          else
+            echo "All required files/keys present! Continuing..."
+          fi
+
+      # Run the sfdx-deploy action w/ conditional args by checking "github" context
+      # NOTE: For list of props see: https://help.github.com/en/actions/reference/contexts-and-expression-syntax-for-github-actions
+      # E.g. find branch/tag name: github.ref returns "refs/heads/my-branch-name"
+      - name: Auth to QA org
+        id: ci-validate-qa
+        if: github.base_ref == 'dev'
+        uses: ./.github/actions/sfdx-validate
+        with:
+          auth-file-key: ${{ secrets.AUTH_FILE_KEY }}
+          target-alias: 'QA'
```

Comment:
In testing, if targeting the "dev" branch but not actually using a sandbox, can comment this out temporarily and it will default to "Prod"